docs/source/examples/classification_and_regression.rst

```diff
@@ -0,0 +1,302 @@
+
+.. code-block:: python
+
+    network = ClassifierLocalizer(model_name='resnet18')
+
+    def freeze_weights(network):
+        for name, param in network.named_parameters():
+            if not name.startswith('fc.'):
+                param.requires_grad = False
+
+    freeze_weights(network)
+    print(network)
+
+.. code-block:: python
+
+    network = ClassifierLocalizer(model_name='resnet18')  # network without freezing any layer.
+
+Loss function
+=============
+
+As we discussed before, we have two different tasks in this example. These tasks need different loss functions; Cross-Entropy loss for the classification and Mean Square Error loss for the regression. Below, we define a new loss function class that sums both losses to considers them simultaneously. However, as the regression is relatively a harder task, we apply a higher weight to MSEloss.
+
+.. code-block:: python
+
+    class ClassificationRegressionLoss(nn.Module):
+        def __init__(self):
+            super(ClassificationRegressionLoss, self).__init__()
+            self.ce_loss = nn.CrossEntropyLoss() # size_average=False
+            self.mse_loss = nn.MSELoss()
+            
+        def forward(self, y_pred, y_true):
+            loss_cls = self.ce_loss(y_pred[0], Variable(y_true[0][:,20])) # Cross Entropy Error (for classification)
+            loss_reg1 = self.mse_loss(y_pred[1][:,0], Variable(y_true[1][:,0]/h, requires_grad=True)) # Mean Squared Error for X_L
+            loss_reg2 = self.mse_loss(y_pred[1][:,1], Variable(y_true[1][:,1]/w, requires_grad=True)) # Mean Squared Error for Y_L
+            loss_reg3 = self.mse_loss(y_pred[1][:,2], Variable(y_true[1][:,2]/h, requires_grad=True)) # Mean Squared Error for X_R
+            loss_reg4 = self.mse_loss(y_pred[1][:,3], Variable(y_true[1][:,3]/w, requires_grad=True)) # Mean Squared Error for Y_R
+            total_loss = loss_cls + loss_reg1 + loss_reg2 + loss_reg3 + loss_reg4
+            return total_loss
+
+Training
+========
+
+.. code-block:: python
+
+    optimizer = optim.Adam(network.parameters(), lr=learning_rate, weight_decay=0)
+    loss_function = ClassificationRegressionLoss()
+    #Step_Learning_Rate = StepLR(step_size=2 , gamma=0.1, last_epoch=-1, verbose=False)
+    exp = Experiment('./two_task_example', network, optimizer=optimizer, loss_function=loss_function, device="all")
+    exp.train(train_dataloader, valid_dataloader, callbacks=callbacks, epochs=num_epochs)
+
+Evaluation
+==========
+
+As you have also noticed from the training logs, we have achieved the best performance (considering the validation loss) at the 15th epoch. The weights of the network for the corresponding epoch have been automatically saved and we use these parameters to evaluate our algorithm visually. Hence,  we take advantage of evaluate function of Poutyne, and apply it to the validation dataset. It provides us the predictions as well as the ground-truth for comparison, in case of need.
```

Comment:
> we have achieved the best performance (considering the validation loss) at the 15th epoch.

Is this always the case? Anyway, I think you could the load_checkpoint function in Experiment that allows to load the best checkpooint.

Reply:
No, i will change it to clearly say that in this run (for example) it happened in the 15th epoch and the Poutyne automatically saves the best weights (whether it is in 15 to 14, or...). I also used evaluate function to access predictions and expectations simultaneously.

Reply:
I'm not sure it's the right thing to do because if someone run the notebook, it will not work and will have to manually change the filename in the code.

Reply:
I changed it to select checkpoint which contains the best weights without mentioning the epoch number directly. i think this file contains the best weights, doesn't it?

Reply:
checkpoint.ckpt does not contain the best checkpoint but the last checkpoint. You should really use the load_checkpoint method in Experiment since it is there for that. You can pass it `'best'` to load the best checkpoint. Also, you can use the test method in Experiment instead of the evaluate_generator in Model to do the same thing you're currently doing. Or, you can get the Model inside the Experiment via `expt.model`.